samples/bluetooth/nfc_pairing/src/main.c

```diff
@@ -0,0 +1,477 @@
+static int tk_value_generate(void)
+{
+	int err;
+	struct device *dev;
+
+	dev = device_get_binding(DT_CHOSEN_ZEPHYR_ENTROPY_LABEL);
+	if (!dev) {
+		printk("error: no entropy device\n");
+		return -ENXIO;
+	}
+
+	err = entropy_get_entropy(dev, tk_value, sizeof(tk_value));
+	if (err) {
+		printk("entropy_get_entropy failed: %d\n", err);
+	}
+
+	return err;
+}
+
+static void bond_find(const struct bt_bond_info *info, void *user_data)
+{
+	int err;
+
+	/* Filter already connected peers. */
+	for (size_t i = 0; i < CONFIG_BT_MAX_CONN; i++) {
+		if (bt_conn_lookup_addr_le(BT_ID_DEFAULT, &info->addr)) {
+			return;
+		}
+	}
+
+	err = k_msgq_put(&bonds_queue, (void *) &info->addr, K_NO_WAIT);
+	if (err) {
+		printk("No space in the queue for the bond.\n");
+	}
+}
+
+static void advertising_start(void)
+{
+	k_msgq_purge(&bonds_queue);
+	bt_foreach_bond(BT_ID_DEFAULT, bond_find, NULL);
+
+	k_work_submit(&adv_work);
+}
+
+/** .. include_startingpoint_pair_msg_rst */
+static int pairing_msg_generate(u32_t *len)
+{
+	int err;
+	struct nfc_ndef_le_oob_rec_payload_desc rec_payload;
+
+	err = bt_le_oob_get_local(BT_ID_DEFAULT, &oob_local);
+	if (err) {
+		printk("Error while fetching local OOB data: %d\n", err);
+	}
+
+	err = tk_value_generate();
+	if (err) {
+		return err;
+	}
+
+	memset(&rec_payload, 0, sizeof(rec_payload));
+
+	rec_payload.addr = &oob_local.addr;
+	rec_payload.le_sc_data = &oob_local.le_sc_data;
+	rec_payload.tk_value = tk_value;
+	rec_payload.local_name = bt_get_name();
+	rec_payload.le_role = NFC_NDEF_LE_OOB_REC_LE_ROLE(
+		NFC_NDEF_LE_OOB_REC_LE_ROLE_PERIPH_ONLY);
+	rec_payload.appearance = NFC_NDEF_LE_OOB_REC_APPEARANCE(
+		CONFIG_BT_DEVICE_APPEARANCE);
+	rec_payload.flags = NFC_NDEF_LE_OOB_REC_FLAGS(BT_LE_AD_NO_BREDR);
+
+	NFC_NDEF_LE_OOB_RECORD_DESC_DEF(oob_rec, '0', &rec_payload);
+	NFC_NDEF_CH_AC_RECORD_DESC_DEF(oob_ac, NFC_AC_CPS_ACTIVE, 1, "0", 0);
+
+	return nfc_ndef_ch_msg_hs_encode(&NFC_NDEF_CH_AC_RECORD_DESC(oob_ac),
+					 &NFC_NDEF_LE_OOB_RECORD_DESC(oob_rec),
+					 1, ndef_msg_buf, len);
+}
+/** .. include_endpoint_pair_msg_rst */
+
+/**
+ * @brief Callback function for handling NFC events.
+ */
+static void nfc_callback(void *context,
+			 enum nfc_t4t_event event,
+			 const u8_t *data,
+			 size_t data_length,
+			 u32_t flags)
+{
+	ARG_UNUSED(context);
+	ARG_UNUSED(data);
+	ARG_UNUSED(flags);
+
+	switch (event) {
+	case NFC_T4T_EVENT_FIELD_ON:
+		dk_set_led_on(NFC_FIELD_LED);
+		break;
+
+	case NFC_T4T_EVENT_FIELD_OFF:
+		dk_set_led_off(NFC_FIELD_LED);
+		break;
+
+	case NFC_T4T_EVENT_NDEF_READ:
+		advertising_start();
+
+		break;
+
+	default:
+		break;
+	}
+}
+
+static void pair_msg_handler(struct k_work *work)
+{
+	int err;
+	u32_t len = sizeof(ndef_msg_buf);
+
+	err = pairing_msg_generate(&len);
+	if (err) {
+		printk("Paring message encode error\n");
+	}
+}
+
+static void advertising_continue(void)
+{
+	struct bt_le_adv_param adv_param;
+
+	bt_addr_le_t addr;
+
+	if (!k_msgq_get(&bonds_queue, &addr, K_NO_WAIT)) {
```

Comment:
If `CONFIG_BT_MAX_PAIRED > 1` we will start directed advertising to the first peer we are bonded with?

Reply:
Yes we started advertising to first bonded peer which is not connected. When dir adv failed or there is empty connection slot, advertising will be started to next bonded peer.